lib/browser/api/web-contents.ts

```diff
@@ -434,6 +434,28 @@ WebContents.prototype.loadFile = function (filePath, options = {}) {
   }));
 };
 
+WebContents.prototype.setWindowOpenOverride = function (handler: (details: { url: string, frameName: string }) => BrowserWindowConstructorOptions | boolean) {
+  this.windowOpenOverrideHandler = handler;
+};
+
+WebContents.prototype._callWindowOpenOverride = function (event: any, url: string, frameName: string) {
+  if (this.windowOpenOverrideHandler) {
+    const response = this.windowOpenOverrideHandler({ url, frameName });
+
+    switch (typeof response) {
+      case 'boolean':
+        if (response === false) event.preventDefault();
+        return;
+      case 'object':
+        return response;
+      default:
+        throw new TypeError(
```

Comment:
This error is uncatchable, we should either just log this or refactor the API so the error is catchable